src/components/Modal/Modal.tsx

```diff
@@ -38,34 +44,67 @@ export const Modal = ({
   title,
   ...wrapperProps
 }: Props): ReactElement => {
+  const focusableElementSelectors =
+    'a[href]:not([tabindex="-1"]), button:not([disabled]), textarea:not([disabled]):not([tabindex="-1"]), input:not([disabled]):not([tabindex="-1"]), select:not([disabled]):not([tabindex="-1"]), area[href]:not([tabindex="-1"]), iframe:not([tabindex="-1"]), [tabindex]:not([tabindex="-1"]), [contentEditable=true]:not([tabindex="-1"])';
   const descriptionId = useRef(nanoid());
   const titleId = useRef(nanoid());
 
-  // This useEffect sets up listeners so the panel will close if user clicks
-  // anywhere else on the page or hits the escape key
+  const modalRef: MutableRefObject<HTMLElement> = useRef(null);
+  const focusableModalElements = useRef(null);
+  const handleTabKey = (e: KeyboardEvent<HTMLDivElement>) => {
```

Comment:
Out of curiosity - is this neccessary to completely override Tabs mechanism in modal?

As far as I remember in example in Vanilla we simply catch tabbing out from last focusable element and redirect focus to first one. Here it seems we are reimplementing tab functionality completely.